components/engine_traits/src/engine.rs

```diff
@@ -153,32 +153,79 @@ impl Drop for TabletErrorCollector {
     }
 }
 
+#[derive(Default)]
+pub struct OpenOptions {
+    // create tablet if non-exist
+    create: bool,
+    create_new: bool,
+    read_only: bool,
+    cache_only: bool,
+}
+
+impl OpenOptions {
+    /// Sets the option to create a new file, or open it if it already exists.
+    pub fn set_create(mut self, create: bool) -> Self {
+        self.create = create;
+        self
+    }
+
+    /// Sets the option to create a new file, failing if it already exists.
+    pub fn set_create_new(mut self, create_new: bool) -> Self {
+        self.create_new = create_new;
+        self
+    }
+
+    /// Sets the option for read only
+    pub fn set_read_only(mut self, read_only: bool) -> Self {
+        self.read_only = read_only;
+        self
+    }
+
+    /// Sets the option for only reading from cache.
+    pub fn set_cache_only(mut self, cache_only: bool) -> Self {
+        self.cache_only = cache_only;
+        self
+    }
+
+    pub fn create(&self) -> bool {
+        self.create
+    }
+
+    pub fn create_new(&self) -> bool {
+        self.create_new
+    }
+
+    pub fn read_only(&self) -> bool {
+        self.read_only
+    }
+
+    pub fn cache_only(&self) -> bool {
+        self.cache_only
+    }
+
+    pub fn validate(&self) -> Result<()> {
```

Comment:
How about overwriting instead of validating? For example, if ignore `cache_only` if `create` is set.

Reply:
Fine.